### ISSUE TITLE
Error in Diff Tracker

### ISSUE THREAD

**Issue by Therzok** [mono/monodevelop#3610]
```
error while DiffTracker.TrackDocument_TextChanged:System.ArgumentOutOfRangeException: Value must be between 0 and 172
Parameter name: index
Actual value was 174.
  at Mono.TextEditor.Utils.CompressingTreeList`1[T].RemoveRange (System.Int32 index, System.Int32 count) [0x00015] in /Users/therzok/Work/md/monodevelop/main/src/core/Mono.TextEditor.Shared/Mono.TextEditor.Utils/CompressingTreeList.cs:202
  at Mono.TextEditor.Utils.CompressingTreeList`1[T].RemoveAt (System.Int32 index) [0x00001] in /Users/therzok/Work/md/monodevelop/main/src/core/Mono.TextEditor.Shared/Mono.TextEditor.Utils/CompressingTreeList.cs:304
  at Mono.TextEditor.Utils.CompressingTreeList`1[T].set_Item (System.Int32 index, T value) [0x00001] in /Users/therzok/Work/md/monodevelop/main/src/core/Mono.TextEditor.Shared/Mono.TextEditor.Utils/CompressingTreeList.cs:314
  at Mono.TextEditor.DiffTracker.TrackDocument_TextChanged (System.Object sender, MonoDevelop.Core.Text.TextChangeEventArgs e) [0x00092] in /Users/therzok/Work/md/monodevelop/main/src/core/Mono.TextEditor.Shared/Mono.TextEditor/Document/DiffTracker.cs:99
```

> VS bug #568080

**Comment by xamarin-release-manager** [comment]
Fixed in version 7.5.0.514 (master)

Pull Request #3954 merged by: Mike Krüger
Author: mono
Commit: 518a1b8d4872cdb71db6321600c32a038b23775d (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.6.0.802 (master)

Pull Request #3954 merged by: Mike Kr??ger
Author: mono
Commit: 518a1b8d4872cdb71db6321600c32a038b23775d (xamarin/monodevelop)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-15] Comment by **xamarinc**



Fixed in version 8.6.0.802 (master)



Pull Request #3954 merged by: Mike Kr??ger  
 Author: mono  
 Commit: 518a1b8d4872cdb71db6321600c32a038b23775d (xamarin/monodevelop)
